### PR TITLE
Fixes to launch Magento with PHP 8.1 - part 2

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
Fixes for compatibility with PHP 8.1 (to be able to launch Magento)
Part 2

### Related Issues
1. magento/magento2#34505

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by xmav** [comment]
@mageto run Static Tests, Semantic Version Checker

**Comment by karyna-tsymbal-atwix** [comment]
@mageto run Static Tests, Semantic Version Checker

**Comment by andrewbess** [comment]
@mageto run Static Tests, Semantic Version Checker

**Review by ihor-sviziev — APPROVED** [review]

**Comment by xmav** [comment]
@magento run Static Tests

**Comment by xmav** [comment]
@magento run Semantic Version Checker

**Review by andrewbess — APPROVED** [review]
Hello @karyna-tsymbal-atwix 
Thank you for your work here.
Approved ✔️

**Comment by andrewbess** [comment]
@magento run Static Tests with env PHP 8.1

**Review by ihor-sviziev — APPROVED** [review]

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests

**Comment by andrewbess** [comment]
@magento run Static Tests

**Comment by andrewbess** [comment]
Hello @karyna-tsymbal-atwix 
Could you please fix Static Tests?
Thank you in advance.

**Comment by karyna-tsymbal-atwix** [comment]
@andrewbess tests were fixed except the "Comment block is missing" warning. This requires to update Magento Coding Standards to ignore #[\ReturnTypeWillChange] after the docblock.

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Static Tests

**Comment by xmav** [comment]
@magento run Static Tests, Semantic Version Checker

File: lib/internal/Magento/Framework/Data/Form/Element/Collection.php [hunk 1]
```diff
@@ -121,6 +126,7 @@ public function add(AbstractElement $element, $after = false)
                 if ($currElement->getId() == $after) {
                     $newOrderElements[] = $currElement;
                     $newOrderElements[] = $element;
+                    // phpcs:ignore Magento2.Performance.ForeachArrayMerge
```
Comment (xmav) [inline lib/internal/Magento/Framework/Data/Form/Element/Collection.php:129]: Should be possible to fix - https://github.com/magento/magento-coding-standard/issues/20

Comment (andrewbess) [reply lib/internal/Magento/Framework/Data/Form/Element/Collection.php:129]: Hello @xmav 
Thank you for your review.
It is not possible here.

**Comment by xmav** [comment]
All tests will be re-checked during delivery to 2.4-develop

**Pull request merged**